src/source/Dtls/Dtls.c

```diff
@@ -508,7 +508,7 @@ STATUS dtlsSessionProcessPacket(PDtlsSession pDtlsSession, PBYTE pData, PINT32 p
         LOG_OPENSSL_ERROR("SSL_read");
     }
 
-    if (!ATOMIC_LOAD_BOOL(&pDtlsSession->sslInitFinished)) {
+    if (ATOMIC_LOAD_BOOL(&pDtlsSession->sslInitFinished)) {
```

Comment:
The if statement looks weird. I am not sure I understand this. So, we are saying that if the initialization is finished we check the outgoing data buffer. However, if it's not then we set the data length? Can you clarify the logic here?